rust/src/http2/http2.rs

```diff
@@ -160,6 +168,46 @@ impl HTTP2Transaction {
         }
     }
 
+    #[cfg(not(feature = "decompression"))]
+    fn handle_headers(&mut self, _blocks: &Vec<parser::HTTP2FrameHeaderBlock>, _dir: u8) {}
+
+    #[cfg(feature = "decompression")]
+    fn handle_headers(&mut self, blocks: &Vec<parser::HTTP2FrameHeaderBlock>, dir: u8) {
+        for i in 0..blocks.len() {
+            if blocks[i].name == "content-encoding".as_bytes().to_vec() {
+                self.decoder.http2_encoding_fromvec(&blocks[i].value, dir);
+            }
+        }
+    }
+
+    // dir may be unused without the decompression feature
+    fn decompress<'a>(
+        &'a mut self, input: &'a [u8], _dir: u8, sfcm: &'static SuricataFileContext, over: bool,
+        files: &mut FileContainer, flags: u16,
+    ) -> io::Result<()> {
```

Comment:
Spent a lot of time in the following lines
`_dir` looks a bit ugly 
We need to define output here so that its lifetime can be used in `self.ft.new_chunk ` the same as `input`

Reply:
`_dir` is fine by me...